processor/metricextractprocessor/config.go

```diff
@@ -50,31 +54,58 @@ const (
 )
 
 var (
-	// errExtractMissing is the error message for a missing extract expression.
-	errExtractMissing = errors.New("extract expression is required")
+	// errExprExtractMissing is the error message for a missing expr extract expression.
+	errExprExtractMissing = errors.New("extract expression is required")
+
+	// errExtractMissing is the error message for a missing ottl extract expression.
+	errOTTLExtractMissing = errors.New("ottl_extract expression is required")
 
 	// errMetricTypeInvalid is the error message for an invalid metric type.
 	errMetricTypeInvalid = errors.New("invalid metric type")
 )
 
 // Config is the config of the processor.
 type Config struct {
-	Route      string            `mapstructure:"route"`
-	Match      string            `mapstructure:"match"`
-	Extract    string            `mapstructure:"extract"`
-	MetricName string            `mapstructure:"metric_name"`
-	MetricUnit string            `mapstructure:"metric_unit"`
-	MetricType string            `mapstructure:"metric_type"`
-	Attributes map[string]string `mapstructure:"attributes"`
+	Route          string            `mapstructure:"route"`
+	Match          *string           `mapstructure:"match"`
```

Comment:
have we alked about how deprecation looks like in this repo yet? are we just following otel standard of 2 releases i think it is? do we have our deperecation schedule anywhere?